Gems/Multiplayer/Code/Source/Components/NetworkDebugPlayerIdComponent.cpp

```diff
@@ -0,0 +1,185 @@
+        m_drawParams.m_hAlign = AzFramework::TextHorizontalAlignment::Center;
+        m_drawParams.m_position = AZ::Vector3(aznumeric_cast<float>(renderScreenpoint.m_x), aznumeric_cast<float>(renderScreenpoint.m_y), 0.f);
+
+        AZStd::string playerIdText = AZStd::string::format("Player %i", GetPlayerId());
+        m_fontDrawInterface->DrawScreenAlignedText2d(m_drawParams, playerIdText);
+
+        if (IsNetEntityRoleAutonomous())
+        {
+            const auto controller = static_cast<NetworkDebugPlayerIdComponentController*>(GetController());
+            if (!controller)
+            {
+                AZ_Assert(false, "NetworkDebugPlayerIdComponent failed to access its multiplayer controller.");
+                return;
+            }
+
+            // Render connection count in the lower right-hand corner
+            const float textHeight = m_fontDrawInterface->GetTextSize(m_drawParams, playerIdText).GetY();
+            const float lineSpacing = 0.5f * textHeight;
+            const AzFramework::WindowSize viewportSize = m_viewport->GetViewportSize();
+            const AZ::Vector3 viewportConnectionBottomRightBorderPadding(-40.0f, -40.0f + textHeight + lineSpacing, 0.0f);
+            m_drawParams.m_hAlign = AzFramework::TextHorizontalAlignment::Right;
+            m_drawParams.m_position =
+                AZ::Vector3(aznumeric_cast<float>(viewportSize.m_width), aznumeric_cast<float>(viewportSize.m_height), 1.0f) +
+                AZ::Vector3(viewportConnectionBottomRightBorderPadding) * m_viewport->GetDpiScalingFactor();
+
+            AZStd::string playerCountText = AZStd::string::format("Player Count: %i", controller->GetConnectionCount());
+            m_fontDrawInterface->DrawScreenAlignedText2d(m_drawParams, playerCountText);
+        }
+    }
+
+    NetworkDebugPlayerIdComponentController::NetworkDebugPlayerIdComponentController(NetworkDebugPlayerIdComponent& parent)
+        : NetworkDebugPlayerIdComponentControllerBase(parent)
+    {
+    }
+
+    void NetworkDebugPlayerIdComponentController::OnActivate([[maybe_unused]] EntityIsMigrating entityIsMigrating)
+    {
+        # if AZ_TRAIT_SERVER
+            m_networkInterface = AZ::Interface<AzNetworking::INetworking>::Get()->RetrieveNetworkInterface(AZ::Name(MpNetworkInterfaceName));
+            if (IsNetEntityRoleAuthority())
+            {
+                uint32_t currentConnectionCount = m_networkInterface->GetConnectionSet().GetConnectionCount();
+                //SetConnectionCount(currentConnectionCount);
+
+                // Multiplayer system doesn't directly track the number of players.
+                // Instead just assign this player an id by checking how many machines are already connected to this host.
+                // Note 1: Players can have duplicate ids based on how many players were connected at the time they activated.
```

Comment:
(maybe a future change) if giving IDs based on connection count is unreliable (i join when there's 3 players so i'm player 4, one leaves, the next person to join is _also_ player 4) what do you think about random assignment of an adjective + noun? like "VibrantEagle", "TerrificSnake", etc. - that way uniqueness is better guaranteed

Reply:
Haha, that would be cool. Definitely more game specific though. I was thinking at the very least broadcast event the MultiplayerAgentDatum (which does actually contain a unique player id) to the player entity.

Reply:
Nice, that sounds like a much more fully-baked idea 😄